### PR TITLE
MXNET-650 [Perl] Gluon ModelZoo, Gluon examples 

### PR DESCRIPTION
**Pull request by sergeykolychev**
## Description ##
1) Added Gluon Contrib modules
2) Added Gluon Model Zoo
3) Added Gluon image classification and style transfer examples
4) Miscellaneous bugfixes and improvements.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
1) Added Gluon Contrib modules
2) Added Gluon Model Zoo
3) Added Gluon image classification and style transfer examples
4) Miscellaneous bugfixes and improvements.

### CODE REVIEW

**Comment by sergeykolychev** [comment]
@tlby 
Robert, could you please review this one, thanks!

File: perl-package/AI-MXNet-Gluon-ModelZoo/examples/image_classification.pl [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+use strict;
+use warnings;
+use AI::MXNet::Gluon::ModelZoo 'get_model';
+use AI::MXNet::Gluon::Utils 'download';
+use Getopt::Long qw(HelpMessage);
+
+GetOptions(
+    ## my Pembroke Welsh Corgi Kyuubi, enjoing Solar eclipse of August 21, 2017
+    'image=s' => \(my $image = 'https://scontent-sea1-1.cdninstagram.com/vp/'.
```
Comment (szha) [inline perl-package/AI-MXNet-Gluon-ModelZoo/examples/image_classification.pl]: 😆 
How long does this pic persist on the CDN of instagram? Will the link expire?

Comment (sergeykolychev) [reply perl-package/AI-MXNet-Gluon-ModelZoo/examples/image_classification.pl]: @szha I am not sure.
But I'll be glad if I can put this image on mxnet data server. Besides that I have a few blog posts that I want in future to have added as a tutorials for perl api
http://blogs.perl.org/users/sergey_kolychev/, they do contain some images and data files that would be nice to have on mxnet data server as well.
I'll be writing new blog post soon announcing ModelZoo for perl and I'll use images from https://www.facebook.com/sergey.kolychev.39/posts/10155737425303434.
Can you help me with putting these images/data files to the server ?
Or may be I can get a write access somehow so I'll not bother you with this ?

Comment (szha) [reply perl-package/AI-MXNet-Gluon-ModelZoo/examples/image_classification.pl]: http://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/kyuubi.jpg

**Comment by sergeykolychev** [comment]
@szha Thank you! I'll update the file with new uri. When you have time could you, please, upload the images from that facebook post as well ? I'll then change stock images with Kyuubi's images for the style transfer example cause they are quite a bit cutier :-)

File: perl-package/AI-MXNet-Gluon-ModelZoo/examples/image_classification.pl (hunk 1 above)
Comment (sergeykolychev) [reply perl-package/AI-MXNet-Gluon-ModelZoo/examples/image_classification.pl]: Thanks!

**Pull request merged**